File: src/main/java/com/example/demo/mapper/WhitelistIpMapper.java
package com.example.demo.mapper;

import com.example.demo.model.WhitelistIp;
import org.apache.ibatis.annotations.Mapper;
import org.apache.ibatis.annotations.Param;

import java.util.List;

@Mapper
public interface WhitelistIpMapper {
    int insert(WhitelistIp ip);

    boolean existsByIp(@Param("ipAddress") String ipAddress);

    List<WhitelistIp> getAll();

    List<WhitelistIp> getActive();

    int markBlacklistedAndIncrement(@Param("ipAddress") String ipAddress);
}

File: src/main/kotlin/com/example/demo/mapper/WhitelistIpMapper.kt
package com.example.demo.mapper

import com.example.demo.model.WhitelistIp
import org.apache.ibatis.annotations.Mapper
import org.apache.ibatis.annotations.Param

@Mapper
interface WhitelistIpMapper {
    fun insert(ip: WhitelistIp): Int
    fun existsByIp(@Param("ipAddress") ipAddress: String): Boolean
    fun getAll(): List<WhitelistIp>
    fun getActive(): List<WhitelistIp>
    fun markBlacklistedAndIncrement(@Param("ipAddress") ipAddress: String): Int
}
